staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -426,7 +413,7 @@ func (s unstructuredJSONScheme) decodeToList(data []byte, list *UnstructuredList
 	itemKind := strings.TrimSuffix(listKind, "List")
 
 	delete(list.Object, "items")
-	list.Items = nil
+	list.Items = make([]Unstructured, 0, len(dList.Items))
```

Comment:
empty `dList.Items` gave nil before, but [] now. Not sure this matters. But I would probably keep the old behaviour to avoid regressions which are hard to find.

Reply:
`UnstructuredList.DeepCopy()` does this today already so should not be a problem:
https://github.com/kubernetes/kubernetes/blob/51e653dc68e47584ab571fc214a7aa5950918831/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L158-L170

I've also checked all places where this field is used - no issues. JSON encoding should not change too - it always produces an empty array:
https://github.com/kubernetes/kubernetes/blob/51e653dc68e47584ab571fc214a7aa5950918831/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L692-L706